akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/EventSourcedBehaviorTestKit.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.scaladsl
+
+import scala.collection.immutable
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.persistence.testkit.internal.EventSourcedBehaviorTestKitImpl
+import akka.persistence.testkit.scaladsl.EventSourcedBehaviorTestKit.CommandResult
+
+object EventSourcedBehaviorTestKit {
+
+  def apply[Command, Event, State](
+      system: ActorSystem[_],
+      behavior: Behavior[Command]): EventSourcedBehaviorTestKit[Command, Event, State] =
+    new EventSourcedBehaviorTestKitImpl(ActorTestKit(system), behavior)
+
+  final case class CommandResult[Command, Event, State, Reply](
+      command: Command,
+      events: immutable.Seq[Event],
+      state: State,
+      reply: Option[Reply])
+}
+
+trait EventSourcedBehaviorTestKit[Command, Event, State] {
+  def runCommand(command: Command): CommandResult[Command, Event, State, Nothing]
+
+  def runCommand[R](creator: ActorRef[R] => Command): CommandResult[Command, Event, State, R]
+}
```

Comment:
Some API suggestion to come closer to the user experience when using `ActorTestKit` and `EntityRef`

We have `tell` and `ask`, instead of `runCommand`.

The `CommandResult` could be: 

```scala 
final class CommandResult[Command, Event, State, Reply](
      val command: Command,
      val state: State,
      private val events: immutable.Seq[Event],
      private val reply: Option[Reply]) {
	
	def expectEvent[ E <: Event](): E
	def expectReply(): Reply
    // and some other synchronous variants
}
```

As it is now, the user has to deal with `List[Event]` and `Option[Reply]`. We can remove the verbose and at the same time have an API that's similar to the `TestProbe`

Reply:
It will be also cool to be able to get a `Source[Event, _]` from the testkit. Then we can feed it to the `ProjectionTestKit`.

Or simply use the `StreamTestKit` with it.

Reply:
there is some other work in progress about adding persistence queries to the inmem journal, then you can easily connect this with projections

Reply:
I have improved `CommandResult`, for example with `eventOfType`, `stateOfType` and `replyOfType`. Also removed the Option for the reply. I agree that it's very common that there is only one event and then no need to juggle the List.

I didn't go as far a s making the CommandResult look like a probe, because I think it's value in keeping it an immutable result and then use other test facilities to assert.

> We have tell and ask, instead of runCommand.

I understand the good intention with the suggestion, but `tell` and `ask` have a very strong association with that it's sending a message. We should not blur that definition by using it for things that are not true async message passing.

`runCommand` has familiarity with `run` in the sync `BehaviorTestKit`.

Reply:
I agree with all the conclusions of this thread.